src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx

```diff
@@ -0,0 +1,149 @@
+})
+
+export const CreateNewArtworkListForm: FC<BoxProps> = (props) => {
+  const { dispatch } = useArtworkListsContext()
+  const { dismiss } = useBottomSheetModal()
+  const [commitMutation] = useCreateNewArtworkList()
+
+  const setRecentlyAddedArtworkList = (result: Result) => {
+    dispatch({
+      type: "SET_RECENTLY_ADDED_ARTWORK_LIST",
+      payload: {
+        internalID: result.internalID,
+        name: result.name,
+      },
+    })
+  }
+
+  const preselectRecentlyAddedArtworkList = (artworkListID: string) => {
+    dispatch({
+      type: "ADD_OR_REMOVE_ARTWORK_LIST_ID",
+      payload: {
+        mode: ArtworkListMode.AddingArtworkListIDs,
+        artworkListID,
+      },
+    })
+  }
+
+  const closeCurrentView = () => {
+    dismiss(ArtworkListsViewName.CreateNewArtworkLists)
+  }
+
+  const handleSubmit = (
+    values: CreateNewArtworkListFormValues,
+    helpers: FormikHelpers<CreateNewArtworkListFormValues>
+  ) => {
+    commitMutation({
+      variables: {
+        input: {
+          name: values.name,
+        },
+      },
+      onCompleted: (data) => {
+        const response = data.createCollection?.responseOrError
+        const artworkList = response?.collection!
+        const result: Result = {
+          name: artworkList.name,
+          internalID: artworkList.internalID,
+        }
+
+        setRecentlyAddedArtworkList(result)
+        preselectRecentlyAddedArtworkList(artworkList.internalID)
+        closeCurrentView()
+
+        helpers.setSubmitting(false)
+      },
+      onError: (error) => {
+        helpers.setFieldError("name", error.message)
+        helpers.setSubmitting(false)
+
+        if (__DEV__) {
+          console.error(error)
+        } else {
+          captureMessage(error?.stack!)
+        }
+      },
+    })
+  }
+
+  return (
+    <Box {...props}>
+      <Formik
+        initialValues={INITIAL_FORM_VALUES}
+        validationSchema={validationSchema}
+        onSubmit={handleSubmit}
+      >
+        {(formik) => {
+          const nameLength = formik.values.name.length
+          const isSaveButtonDisabled = !formik.isValid || nameLength === 0
+
+          return (
+            <>
+              <BottomSheetInput
+                placeholder="Name your list"
+                value={formik.values.name}
+                onChangeText={formik.handleChange("name")}
+                error={formik.errors.name}
+                maxLength={MAX_NAME_LENGTH}
+                onBlur={formik.handleBlur("name")}
+              />
+              <InputRemainingCharactersLabel
+                currentLength={nameLength}
+                maxLength={MAX_NAME_LENGTH}
+              />
+
+              <Spacer y={4} />
+
+              <Button
+                width="100%"
+                block
+                disabled={isSaveButtonDisabled}
+                loading={formik.isSubmitting}
+                onPress={formik.handleSubmit}
+              >
```

Comment:
```suggestion
              <Button
                block
                disabled={isSaveButtonDisabled}
                loading={formik.isSubmitting}
                onPress={formik.handleSubmit}
              >
```

We dont need width="100%" if it has block prop

Reply:
yeah, you're right. good catch!

Reply:
Addressed here: https://github.com/artsy/eigen/pull/8593/commits/4ed1197ac80eafeda103b05cad9314dbf6e9b6ef